sklearn/metrics/cluster/_expected_mutual_info_fast.pyx

```diff
@@ -34,37 +36,34 @@ def expected_mutual_information(contingency, int n_samples):
     # While nijs[0] will never be used, having it simplifies the indexing.
     nijs = np.arange(0, max(np.max(a), np.max(b)) + 1, dtype='float')
     nijs[0] = 1  # Stops divide by zero warnings. As its not used, no issue.
+    nijs_view = nijs
     # term1 is nij / N
-    term1 = nijs / N
+    term1 = nijs / n_samples
     # term2 is log((N*nij) / (a * b)) == log(N * nij) - log(a * b)
     log_a = np.log(a)
     log_b = np.log(b)
     # term2 uses log(N * nij) = log(N) + log(nij)
-    log_Nnij = np.log(N) + np.log(nijs)
+    log_Nnij = np.log(n_samples) + np.log(nijs)
```

Comment:
It might be worth renaming `log_n_samples_nij` since we don't use `N` anymore elsewhere.

Reply:
Actually, I find it long and we still use `N` in some other naming. So this is good enough.